var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -270,7 +278,7 @@ def cmake_args(self):
         for library in (
                 'gsl', 'hdf5', 'p4est', 'petsc', 'slepc', 'trilinos', 'metis',
                 'sundials', 'nanoflann', 'assimp', 'gmsh'):
-            if library in spec:
+            if ('+'+library) in spec:
```

Comment:
p.s. that lead to `metis` being enabled because it's a dependency of `suite-sparse` although we had `~metis` variant.